app/src/main/java/com/epfl/drawyourpath/challenge/DailyGoalViewAdapter.kt

```diff
@@ -50,74 +61,123 @@ class DailyGoalViewAdapter(private val dailyGoal: DailyGoal) :
 
-        displayGoal(viewHolder, goalPos)
+    /**
+     * update the daily goal
+     * @param dailyGoal the new daily goal
+     */
+    fun updateDailyGoal(dailyGoal: DailyGoal) {
+        this.dailyGoal = dailyGoal
+        notifyItemRangeChanged(0, itemCount)
+    }
 
+    /**
+     * close the keyboard
+     * @param view
+     */
+    private fun closeKeyboard(view: View) {
+        val imm = view.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.hideSoftInputFromWindow(view.windowToken, 0)
     }
 
-    // Return the size of the dataset (invoked by the layout manager)
-    override fun getItemCount() = 3
+    /**
+     * set the edit text to DailyGoal
+     * @param viewHolder the view of where the items are
+     * @param position the position inside the RecyclerView
+     */
+    private fun setEditText(viewHolder: ViewHolder, position: GoalPos) {
+        viewHolder.editText.setText(getGoalToDouble(dailyGoal, position).toInt().toString())
+    }
 
     /**
      * display the goal to the view
      *
      * @param viewHolder the view of where the items are
      * @param position the position inside the RecyclerView
      */
-    private fun displayGoal(viewHolder: ViewHolder, position: GoalPos) {
+    private fun displayGoal(viewHolder: ViewHolder, position: GoalPos, context: Context) {
         val currentProgress = getProgressToDouble(dailyGoal, position)
         val goal = getGoalToDouble(dailyGoal, position)
 
-        viewHolder.progressText.text = "$currentProgress/$goal"
+        viewHolder.progressText.text = getProgressText(position, context).format(currentProgress, goal)
         viewHolder.progressBar.max = goal.toInt()
         viewHolder.progressBar.progress = min(currentProgress.toInt(), goal.toInt())
     }
 
     /**
      * get the unit associated with the goal
      * @param pos the position of the goal
-     *
+     * @param context the context of the view (to get the string resource)
      * @return the associated unit
      */
-    private fun getGoalUnit(pos: GoalPos): String {
+    private fun getGoalUnit(pos: GoalPos, context: Context): String {
+        return when (pos) {
+            GoalPos.DISTANCE -> context.getString(R.string.kilometers)
+            GoalPos.TIME -> context.getString(R.string.minutes)
+            GoalPos.PATH -> context.getString(R.string.paths)
+        }
+    }
+
+    /**
+     * get the progress text associated with the goal
+     * @param pos the position of the goal
+     * @param context the context of the view (to get the string resource)
+     */
+    private fun getProgressText(pos: GoalPos, context: Context): String {
         return when (pos) {
-            GoalPos.DISTANCE -> "kilometers"
-            GoalPos.TIME -> "minutes"
-            GoalPos.PATH -> "paths"
+            GoalPos.DISTANCE -> context.getString(R.string.progress_over_goal)
+            GoalPos.TIME -> context.getString(R.string.progress_over_goal)

```

Comment:
I don't think the string extract here with getString are correct same they are the same the time and the distance and it was different before.

Reply:
these are two different function, the one with "minutes" and "paths" is replaced with "R.string.mintes" and "R.string.paths" the other one is used to display 1.000/1 and 1/1 which is a double in the case of the distance and an int for paths.
github displayed the two at the same place but they are not the same function.